stdlib/public/SwiftShims/RefCount.h

```diff
@@ -1329,6 +1330,9 @@ class HeapObjectSideTableEntry {
 #endif
   { }
 
+  void *operator new(size_t) = delete;
```

Comment:
I might have to pull these lines out, but the idea is to catch any errant uses of `new` or `delete` on this type.